WordPress/Classes/Models/AbstractPost.m

```diff
@@ -583,6 +592,32 @@ - (void)markRemoteStatusFailed
     [self save];
 }
 
+- (BOOL)shouldAttemptAutoUpload {
+    if (!self.confirmedChangesTimestamp || !self.confirmedChangesHash) {
+        return NO;
+    }
+
+    NSTimeInterval timeDifference = [[NSDate date] timeIntervalSinceDate:self.confirmedChangesTimestamp];
+
+    BOOL timeDifferenceWithinRange = timeDifference <= (60 * 60 * 24 * 2);
+    // I know, I know. This isn't how a date comparison _should_ be done!
+    // However, going thru NSCalendar can get really expensive and this method can potentially be called a lot during
+    // scrolling of a Post List — and for our specific use-case, being slightly innacurate here in terms of
+    // leap seconds or other calendrical oddities doesn't actually matter.
+
+    BOOL hashesEqual = [self.confirmedChangesHash isEqualToString:[self calculateConfirmedChangesContentHash]];
+
+    return hashesEqual && timeDifferenceWithinRange;
+}
+
+- (void)setShouldAttemptAutoUpload:(BOOL)shouldAttemptAutoUpload {
```

Comment:
While testing #12358 which uses this as the base branch, I found that the Cancel button in the Post List stopped working. I think this is the cause. We're not using the `shouldAttemptAutoUpload` parameter. 🙂

Reply:
🤦‍♂ I don't know how I missed that.